source/simulator/core.cc

```diff
@@ -1042,17 +1042,38 @@ namespace aspect
                mpi_communicator);
 #endif
 
-    DoFTools::make_sparsity_pattern (dof_handler,
-                                     coupling, sp,
-                                     constraints, false,
-                                     Utilities::MPI::
-                                     this_mpi_process(mpi_communicator));
     if ((parameters.use_discontinuous_temperature_discretization) || (parameters.use_discontinuous_composition_discretization))
-      DoFTools::make_flux_sparsity_pattern (dof_handler,
-                                            sp,
-                                            constraints, false,
-                                            Utilities::MPI::
-                                            this_mpi_process(mpi_communicator));
+      {
+        Table<2,DoFTools::Coupling> face_coupling (introspection.n_components,
+                                                   introspection.n_components);
+        const typename Introspection<dim>::ComponentIndices &x
+          = introspection.component_indices;
+        if (parameters.use_discontinuous_temperature_discretization)
+          face_coupling[x.temperature][x.temperature] = DoFTools::always;
+
+        if (parameters.use_discontinuous_composition_discretization)
+          {
+            for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+              face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::always;
+          }
+        /*
+                DoFTools::make_sparsity_pattern (dof_handler,
+                                               coupling, sp,
+                                               constraints, false,
+                                               Utilities::MPI::
+                                               this_mpi_process(mpi_communicator));
+        */
+        DoFTools::make_flux_sparsity_pattern (dof_handler,
+                                              sp,
+                                              coupling,
+                                              face_coupling);
```

Comment:
One issue for parallel is that the function call DoFTools::make_flux_sparsity_pattern doesn't accept the arguments "constraints", "coupling" and "this_mpi_process(mpi_communicator)"  at the same time.